### ISSUE TITLE
Add support for Keystone Trust (v3)

### ISSUE THREAD

**Issue by codevulture** [kubernetes/kubernetes#25862]
Currently no option is available to add Keystone Trust in Kubernetes.

An initial patch to add keystone trust is available on https://github.com/rackspace/gophercloud/pull/580
Diff:
https://github.com/codevulture/gophercloud/commit/149babe7190ed46e8c5c742ba66d896a8811b4c2

**Comment by lavalamp** [comment]
I don't understand what you're asking for, can you clarify?

**Comment by codevulture** [comment]
Hi,
Currently there is no option to use Openstack-Keystone Trust in Kubernetes, Keystone Trust was introduced in Api version 3 and allows a Trustor to delegate roles to Trustee, A more detailed information are provided in the link below:
https://wiki.openstack.org/wiki/Keystone/Trusts

Specifications are provided in: https://specs.openstack.org/openstack/keystone-specs/api/v3/identity-api-v3-os-trust-ext.html
So to add support for that , it needs to be implemented in Gophercloud library first as i have provided a link above and once it is done add that in Kubernetes.

**Comment by anguslees** [comment]
@lavalamp: This is an OpenStack-specific feature that would allow the OpenStack cloud provider to work with a set of credentials that only have limited privileges rather than the full username+password.  It's a good thing :)

It seems I don't have the power to add an `area/platform/openstack` label to this issue - it would be great if someone who was suitably blessed could do so.

**Comment by mkumatag** [comment]
I don't really understand what help is needed here! I already see a WIP PR - https://github.com/kubernetes/kubernetes/pull/26037 out for review. The only thing we need to do here is wait for rackspace/gophercloud#580 to merge and merge kubernetes PR(https://github.com/kubernetes/kubernetes/pull/26037)

**Comment by lavalamp** [comment]
Added label.

**Comment by xsgordon** [comment]
Minor update for future time travellers, https://github.com/kubernetes/kubernetes/pull/26037 has been superseded by https://github.com/kubernetes/kubernetes/pull/32111
